Marlin/src/module/tool_change.cpp

```diff
@@ -954,7 +954,7 @@ void fast_line_to_current(const AxisEnum fr_axis) { _line_to_current(fr_axis, 0.
       #if ENABLED(TOOLCHANGE_PARK)
         if (ok) {
           #if ENABLED(TOOLCHANGE_NO_RETURN)
-            destination.set(current_position.x, current_position.y);
+            destination = current_position;
```

Comment:
It looks to me like the Z position here is supposed to be the original Z position from line 908 above so that Z is moving back down to the previous position here. But maybe that is not meant to apply to `TOOLCHANGE_NO_RETURN`. In any case, simply setting `destination` to `current_position` here would seem to result in `prepare_internal_move_to_destination` doing nothing, so clearly some kid of motion was intended here.

The flag was introduced in #14117 by @InsanityAutomation so we can look closer at that to see what was the original intention.

Reply:
See also #21564 by @espr14 for the change from `do_blocking_move_to_z` to `prepare_internal_move_to_destination`.